pkg/vfs/vfs.go

```diff
@@ -94,6 +94,9 @@ type Indexer interface {
 
 	// CreateFileDoc creates and add in the index a new file document.
 	CreateFileDoc(doc *FileDoc) error
+	// CreateNamedFileDoc creates and add in the index a new file document with
+	// its id already set.
+	CreateNamedFileDoc(doc *FileDoc) error
```

Comment:
I asked what would be the best solution to do so and the reply I got was to augment the interface with two dedicated methods. Hence this.
I can also do what you just said both solutions are fine to me.

Regarding the collisions this is an issue I was not aware of: what you mean is that two instances cannot have a document with the same id? And the collision would occurr at which level? CouchDB or the underlying filesystem?